### PR TITLE
feat: add empty logger for prod use

### PR DESCRIPTION
**Pull request by drev74**
This PR adds a default empty logger fot client. It's annoying to have choria messages in the application code.
Altho it is possible to pass a custom logger, it's inconvenient. 

If I wanna have a `DEBUG` level for my client code, but wanna have a silent AJC, I need to instantiate two loggers with diff levels of verbosity. Otherwise, I'm getting too much noise in console with messages from AJC.

Adding a default empty logger solves this problem. AJC gets silent, but it is still possible to pass a custom logger

### CODE REVIEW

**Comment by ripienaar** [comment]
Please see https://github.com/choria-io/.github/blob/master/CONTRIBUTING.md for guidance on issues, commit formatting etc

File: client.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ func NewClient(opts ...ClientOpt) (*Client, error) {
 		replicas:    1,
 		concurrency: 10,
 		retryPolicy: RetryDefault,
-		logger:      &defaultLogger{},
+		logger:      &emptyLogger{},
```
Comment (ripienaar) [inline client.go:33]: maybe call it `noopLogger`

Comment (drev74) [reply client.go]: Yes, renamed to `noopLogger`

**Comment by drev74** [comment]
solves #85

**Comment by ripienaar** [comment]
Please review the contributing guide for commit formats etc

**Comment by drev74** [comment]
I don't understand the issus. I'm using a standard VSCode auto formatting tool, compatible with vet

**Comment by ripienaar** [comment]
first line should be `(#85) adds noop logger and default to it` followed by your usual commit message, ideally not exceeding 80 chars wide for github display.

Here `#85` is your issue number, then squash to one commit.

**Comment by drev74** [comment]
I updated my commit to meet your requirement. I messed a bit git with an extra merge, sorry, not very good in git

**Pull request merged**